WordPress/Classes/Services/PostService.m

```diff
@@ -271,6 +270,62 @@ - (void)uploadPost:(AbstractPost *)post
     }
 }
 
+- (void)savePost:(AbstractPost *)post
+         success:(nullable void (^)(AbstractPost *post, NSString *previewURL))success
+         failure:(void (^)(NSError * _Nullable error))failure
+{
+    if(![post.blog supports:BlogFeatureWPComRESTAPI]) {
+        NSDictionary *userInfo = @{ NSLocalizedDescriptionKey : @"Blog does not support WPCom REST API" };
+        failure([NSError errorWithDomain:PostServiceErrorDomain code:0 userInfo:userInfo]);
+        return;
+    }
+
+    id<PostServiceRemote> remote = [self remoteForBlog:post.blog];
+    RemotePost *remotePost = [self remotePostWithPost:post];
+    NSManagedObjectID *postObjectID = post.objectID;
+    void (^successBlock)(RemotePost *post, NSString *previewURL) = ^(RemotePost *post, NSString *previewURL) {
+        [self.managedObjectContext performBlock:^{
+            AbstractPost *postInContext = (AbstractPost *)[self.managedObjectContext existingObjectWithID:postObjectID error:nil];
+            if (postInContext) {
+                postInContext.remoteStatus = AbstractPostRemoteStatusSaved;
+                NSPredicate *unattachedMediaPredicate = [NSPredicate predicateWithFormat:@"postID <= 0"];
```

Comment:
The unattached media handling is almost identical between the `savePost` and `updatePost` methods, the only difference is the parameters in the success blocks.   I wonder if we could refactor this to its own method to avoid the duplication?